### ISSUE TITLE
[BUG] Apple rejection because of Apple Pay icon

### ISSUE THREAD

**Issue by keoztiryaki** [stripe/stripe-ios#2080]
## Summary
Two of our apps that use Stripe is now getting rejected by Apple due to "using not approved Apple Pay icons".

## Code to reproduce
We open Stripe's prebuild ui of payment methods via paymentContect.presentPaymentOptionsViewController()

## iOS version
v16.1

## Installation method
Cocoapods

## SDK version
v22.8.3 (We can't upgrade to v23 since Stripe cut support for iOS 12)

## Other information
We are using Stripe's prebuilt ui. In this screen, users are able to either choose Apple Pay or add/edit their credit cards. We have been using this for years and yesterday we got 2 rejections from Apple. They said we are using not approved icons of Apple Pay. They also suggested we take a look at the following link for approved icons of Apple Pay. https://developer.apple.com/design/human-interface-guidelines/technologies/apple-pay/buttons-and-marks

We tried saying it is Stripe's ui but they rejected us again. (It was a long shot anyway). We don't have any Apple Pay icon within the project. And they rejected with this reason and sent us the screen that has this problem. I am attaching these screenshots so that you can see Apple's full message.

Obviously the ideal solution would be Stripe using the approved Apple Pay icon. But we are open to any suggestion that could help us release our applications.

![Apple Rejection](https://user-images.githubusercontent.com/19761375/200817269-8b5cc2a0-4e98-49cc-b5d5-bc828ca0d9c9.png)

**Comment by eurias-stripe** [comment]
Hi @keoztiryaki, thanks for reporting this issue.
Can you share a little more information here:
- After tapping the Apple Pay cell in the stripe view controller, do you directly show the Apple Pay sheet?

If that's the case that might be the reason for the rejection, Apple might be getting more strict on this. 
One possible solution would be that instead of directly showing the Apple Pay sheet you show your checkout screen with a `PKPaymentButton` which then shows the Apple Pay sheet.
Another solution would be to use the `PaymentSheet` which is our recommended integration, this does use a `PKPaymentButton` directly and should have no problem going through Apple review, you can find more information about it [here](https://stripe.com/docs/payments/accept-a-payment?platform=ios).

**Comment by keoztiryaki** [comment]
Hello,

Apple is definitely getting more strict about this. Because our flow has been same for a very long time. Every time we submit a new app, they always asked us for Apple Pay functionality, we explained, and every time it was approved. (We have lots of clients and lots of apps using the same code base).

The flow is as following: In our app, we have a "wallet" screen. To add money to the wallet, users must have a valid payment method. Therefore, we open this screen (Stripe's screen). In here, users either choose Apple Pay or add / edit their credit card. Then users go back to their wallet. From that moment forward, when they want to add money to their wallet, we use this preselected payment option. For the first time they add a payment method, we are charging 1 € to save the payment option.

To answer your question: If user is adding this payment method for **the first time**, after tapping Apple Pay, this payment option screen is getting closed, and user is redirected to wallet screen. There, we show a message that we are going to charge 1€. If user approves that, we charge 1 € which means Apple Pay sheet appears.

We do not have checkout screen unfortunately. The flow is like above. We have wallet screen where users can see their balance or add a new balance to their wallet or add a payment option.

**Comment by keoztiryaki** [comment]
Hello,

We tried doing that before we write to here or Stripe's customer support. They rejected again saying that previously stated issue is not fixed.

Here are some screenshots of our app. Maybe it could help you guys.

![1](https://user-images.githubusercontent.com/19761375/201295724-b2149acf-4e16-4e65-89cd-a9289e54101d.png)
<img width="562" alt="2" src="https://user-images.githubusercontent.com/19761375/201295730-7f2f0cdd-0cad-47e4-8103-202e5746982e.png">
![3](https://user-images.githubusercontent.com/19761375/201295750-4544cb0a-c8a4-4fa2-a2d0-5296e7fcbbce.png)

**Comment by keoztiryaki** [comment]
Hello again,

After carefully reviewing the given guideline, we detected that Apple specifically tells developers to use this Apple Pay icon when showing payment options. We appealed to the case saying that we are doing exactly what the guideline says and this is not a button, it is a tableViewController which presents users all available payment options.

Finally they decided to approve. Thanks for your help.


<img width="250" alt="1" src="https://user-images.githubusercontent.com/19761375/201353000-64919c6a-3a1f-4355-b378-c4871e1536be.png">
